### PR TITLE
Fix filters

### PR DESCRIPTION
**Pull request by muselesscreator**
**TL;DR -** 
Fix Filters to appropriately trigger URI changes.
Also started splitting out StudentGroups and CourseGradeFilters.

JIRA: 
[JIRA-5696](https://openedx.atlassian.net/browse/JIRA-5696)
[JIRA-5665](https://openedx.atlassian.net/browse/JIRA-5665) (Partial)
[JIRA-5666](https://openedx.atlassian.net/browse/JIRA-5666) (Partial)

**What changed?**

**Developer Checklist**
- [ ] Test suites passing
- [ ] Received code-owner approving review
- [ ] Bumped version number [package.json](../package.json)

**Testing Instructions**
Pull up gradebook and play with filters.  All filters should immediately update the URI.
Also everything else should still work...

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

**Comment by nsprenkle** [comment]
Is it a small enough task to isolate this just to the bug fix? I find it pretty messy to mix several streams of work (especially partial/in-process) especially when doing blames/rollbacks.

**Review by nsprenkle — CHANGES_REQUESTED** [review]
See comment about isolating bug behavior
